### DISCUSSION TITLE
[Script request] Autoscale VM/LXC

### DISCUSSION THREAD

**Discussion by deku-m** [community-scripts/ProxmoxVE#128] [Request script]
Hi,

I came across these belows:
**- https://github.com/fabriziosalmi/proxmox-lxc-autoscale**
- LXC AutoScale is a resource management daemon specifically designed for Proxmox environments. It automatically adjusts CPU and memory allocations with no downtime and can clone LXC containers based on real-time usage metrics and predefined thresholds. Can be run locally or remotely to make your containers always optimized for performance, managing spikes in demand, and optionally preserving resources during off-peak hours.

**- https://github.com/fabriziosalmi/proxmox-vm-autoscale**
- Proxmox VM Autoscale is a dynamic scaling service that automatically adjusts virtual machine (VM) resources (CPU cores and RAM) on your Proxmox Virtual Environment (VE) based on real-time metrics and user-defined thresholds. This solution helps ensure efficient resource usage, optimizing performance and resource availability dynamically.
The service supports multiple Proxmox hosts via SSH connections and can be easily installed and managed as a systemd service for seamless automation.

Might this be interesting to implement?

**Comment by MickLesk** [comment]
I know the project. Ideally, however, it would be perfect if we could get him to merge it directly into the community project. I don't want to bend his code and maintain it, he knows what he's doing, so that would probably be the best solution.


The problem is rather how to contact him? Discussion? Issue?

@fabriziosalmi

**Reply by fabriziosalmi** [reply]
Can you kindly suggest me how to do that? Can be a nice way to make such projects be alive for a longer time. TY in advance

**Reply by MickLesk** [reply]
> Can you kindly suggest me how to do that? Can be a nice way to make such projects be alive for a longer time. TY in advance

Hi 😄 

Nice that you reacted directly.

You can create a PR with the script and we can work through it together. Or you can take a look at the scripts in /misc, e.g. “update-lxc”, “kernel-clean”, “pve-post-install” or similar. You would have to adapt your shell to the way these scripts are, e.g. “autoscale-lxc.sh”. 

It would probably make even more sense, since there is something more to your project, to create a new folder \misc\autoscale and store the data there. I have to take a rough look at the scripts first, I can also prepare them next week so that you only have to check and release them if necessary. I was only able to take a quick look at the source code today. 😆

**Reply by fabriziosalmi** [reply]
Thank You for your response. To me the easier is better :) 

I can try to document each function better then your review will be faster and more insighful for all of us, is this sounds good to you? I have no much time since i've 1yo daughter and she is more cpu consuming than any 405B LLM :D

**Reply by droltr** [reply]
https://github.com/droltr/lxc-auto-scaler @fabriziosalmi tx.

**Reply by fabriziosalmi** [reply]
@droltr if you can handle also the vm autoscale scripting same way to link on this repo it will be perfect. Then i can go coding only on source projects for security or impro realted only stuff :)

**Reply by droltr** [reply]
@fabriziosalmi, I noticed a request for a Helper Script, so I generated a code for automated Proxmox installation. The compatibility with Helper Scripts should be verified. I’m not at the level to write something as advanced as yours yet. :)

**Comment by fabriziosalmi** [comment]
working on that ☕

**Comment by droltr** [comment]
@MickLesk,

https://github.com/droltr/lxc-auto-scaler is that usefull for Helper Scripts?

**Comment by fabriziosalmi** [comment]
Finally :) 

Hi @MickLesk! Just opened PR [#10691](https://github.com/community-scripts/ProxmoxVE/pull/10691) with both tools.

Placed them in [pve](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) following the existing host utility pattern. Using Python venv in [opt](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/workbench.html) to handle Debian 12's PEP 668 restrictions cleanly.

Let me know if any adjustments are needed!
